src/components/Meme.jsx

```diff
@@ -0,0 +1,66 @@
+import { useState, useEffect } from "react"
+export default function Meme() {
+    const [meme, setMeme] = useState({
+        topText: "",
+        bottomText: "",
+        randomImage: "http://i.imgflip.com/1bij.jpg" 
+    })
+    const [allMemes, setAllMemes] = useState([])
+    
+    useEffect(() => {
+        fetch("https://api.imgflip.com/get_memes")
+            .then(res => res.json())
+            .then(data => setAllMemes(data.data.memes))
+    }, [])
```

Comment:
You forgot the slight detail of invoking the `clean-up function` to prevent memory leaks.